### DISCUSSION TITLE
cWORD and cword not showing name in UI

### DISCUSSION THREAD

**Discussion by benfrain** [ibhagwan/fzf-lua#1518] [Q&A]
Just went to make another few mappings for FZF but they don't seem to be working quite as expected but not sure if I just misunderstand.

I have this command:

```lua
km.set("n", "<leader>8", require("fzf-lua").grep_cWORD, { desc = "FZF Selection" })
```
And I expected, due to https://github.com/ibhagwan/fzf-lua/issues/29 to see the word in the UI next to the main `RG>`, but I don't. I see it below in 'Grep string'. Is that expected behaviour? 

Same thing for `cword`

FZF v0.55.0, latest stable Neovim and version of this plugin

**Comment by ibhagwan** [comment]
This isn’t a bug, cword uses word boundary regex `/b` and cWORD uses a different regex to attempt replicate the word boundary behavior, it’s obviously not going to meet all cases, press `ctrl-g` to edit the regex to something that works for you, alternatively create a custom bind that expands cword without the `/b`’s.

The code is rather simple if you wish to create your own binds for cword with different (or no regex boundaries):
https://github.com/ibhagwan/fzf-lua/blob/ce1e24f2a48089b65deefcb0a71a9856839002b5/lua/fzf-lua/providers/grep.lua#L373-L387

**Reply by benfrain** [reply]
Thanks for the response. FZF-LUA is finding the correct things in the results when I run the commands, just the input area in the UI is blank, that was all I meant, sorry, not sure I explained that well (I can't paste a grab currently but will later).

**Reply by ibhagwan** [reply]
> Thanks for the response. FZF-LUA is finding the correct things in the results when I run the commands, just the input area in the UI is blank, that was all I meant, sorry, not sure I explained that well (I can't paste a grab currently but will later).

You mean it’s empty as you type? That’s under the control of Fzf but perhaps you have an issue with highlights making it transparent? How does it look like if your run Fzf inside the neovim terminal.

Reading your issue again I think I perhaps understand what you  mean, when you grep cword the UI opens as if you ran `:FzfLua grep search=<WORD>` which sets the underlying regex and opens in “fuzzy” mode, you’re not going to see it in the UI as live_grep but you can see it in the header:
![IMG_2675](https://github.com/user-attachments/assets/5411346d-3a6f-4a1c-8a16-8aa68270588c)

**Reply by benfrain** [reply]
Yes, so in your screen grab, I was expecting to have the bit where the cursor is 'pre-filled' with 'bpcall_expand'

**Reply by ibhagwan** [reply]
> Yes, so in your screen grab, I was expecting to have the bit where the cursor is 'pre-filled' with 'bpcall_expand'

Then it wouldn’t be `grep` (it would be considered “live”), if that’s what you want create your own versions of `live_grep_cword`, use the code from here and change grep to live_grep (L378):
https://github.com/ibhagwan/fzf-lua/blob/ce1e24f2a48089b65deefcb0a71a9856839002b5/lua/fzf-lua/providers/grep.lua#L373-L379

**Reply by benfrain** [reply]
Sorry, I'm not very experienced in this stuff, do you mean add that duped function in that providers/grep.lua file? Or have a function like that in my own config, the kind of thing here: https://github.com/ibhagwan/fzf-lua/wiki/Advanced#live-queries-fzf_live

**Reply by ibhagwan** [reply]
> Sorry, I'm not very experienced in this stuff, do you mean add that duped function in that providers/grep.lua file? Or have a function like that in my own config, the kind of thing here: https://github.com/ibhagwan/fzf-lua/wiki/Advanced#live-queries-fzf_live

Duplicate the fzf-lua code I linked and change from grep to live_grep.
